tests/tuple-tree-generator/typescript/run-ts-roundtrip-test.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+set -euo pipefail
+
+CURRENT_DIR=$( dirname -- "${BASH_SOURCE[0]}" )
+cd -- "$CURRENT_DIR"
+
+MODEL_TMP_FILE=$(mktemp -p "$CURRENT_DIR" --suffix=.in.yml)
+OUT_MODEL_TMP_FILE=$(mktemp -p "$CURRENT_DIR" --suffix=.out.yml)
+
+FILE="$1"
+
+revng model opt -Y --verify < "$FILE" > "$MODEL_TMP_FILE"
+node out/test.js "$MODEL_TMP_FILE" "$OUT_MODEL_TMP_FILE"
+if ! revng model diff "$MODEL_TMP_FILE" "$OUT_MODEL_TMP_FILE"; then
+  echo "Model of $FILE did not reserialize as original"
+  exit 1;
+fi
+
+rm "$MODEL_TMP_FILE" "$OUT_MODEL_TMP_FILE"
```

Comment:
Cleanup not happening in case of failure

Reply:
Fixed